sparse/mlir_backend/tests/test_simple.py

```diff
@@ -108,93 +123,109 @@ def test_dense_format(dtype, shape):
+
+
 @parametrize_dtypes
-def test_2d_constructors(rng, dtype):
+@parametrize_scipy_fmt
+def test_roundtrip(rng, dtype, format):
     SHAPE = (80, 100)
     DENSITY = 0.6
     sampler = generate_sampler(dtype, rng)
-    csr = sps.random_array(SHAPE, density=DENSITY, format="csr", dtype=dtype, random_state=rng, data_sampler=sampler)
-    csc = sps.random_array(SHAPE, density=DENSITY, format="csc", dtype=dtype, random_state=rng, data_sampler=sampler)
-    dense = np.arange(math.prod(SHAPE), dtype=dtype).reshape(SHAPE)
-    coo = sps.random_array(SHAPE, density=DENSITY, format="coo", dtype=dtype, random_state=rng, data_sampler=sampler)
-    coo.sum_duplicates()
+    sps_arr = sps.random_array(
+        SHAPE, density=DENSITY, format=format, dtype=dtype, random_state=rng, data_sampler=sampler
+    )
 
-    csr_tensor = sparse.asarray(csr)
-    csc_tensor = sparse.asarray(csc)
-    dense_tensor = sparse.asarray(dense)
-    coo_tensor = sparse.asarray(coo)
-    dense_2_tensor = sparse.asarray(np.arange(100, dtype=dtype).reshape((25, 4)) + 10)
+    sp_arr = sparse.asarray(sps_arr)
+    sps_roundtripped = sparse.to_scipy(sp_arr)
+    assert_sps_equal(sps_arr, sps_roundtripped)
+
+    sp_arr_roundtripped = sparse.asarray(sps_roundtripped)
 
-    csr_retured = sparse.to_scipy(csr_tensor)
-    assert_sps_equal(csr_retured, csr)
+    assert_array_equal(sp_arr, sp_arr_roundtripped)
 
-    csc_retured = sparse.to_scipy(csc_tensor)
-    assert_sps_equal(csc_retured, csc)
 
-    dense_returned = sparse.to_numpy(dense_tensor)
-    np.testing.assert_equal(dense_returned, dense)
+@parametrize_dtypes
+@pytest.mark.parametrize("shape", [(80, 100), (200,), (10, 20, 30)])
+def test_roundtrip_dense(rng, dtype, shape):
+    sampler = generate_sampler(dtype, rng)
+    np_arr = sampler(shape)
+
+    sp_arr = sparse.asarray(np_arr)
+    np_roundtripped = sparse.to_numpy(sp_arr)
+    assert np_arr.dtype == np_roundtripped.dtype
+    np.testing.assert_array_equal(np_arr, np_roundtripped)
 
-    coo_returned = sparse.to_scipy(coo_tensor)
-    np.testing.assert_equal(coo_returned.todense(), coo.todense())
+    sp_arr_roundtripped = sparse.asarray(np_roundtripped)
 
-    dense_2_returned = sparse.to_numpy(dense_2_tensor)
-    np.testing.assert_equal(dense_2_returned, np.arange(100, dtype=dtype).reshape((25, 4)) + 10)
+    assert_array_equal(sp_arr, sp_arr_roundtripped)
 
 
 @parametrize_dtypes
-def test_add(rng, dtype):
+@parametrize_scipy_fmt
+def test_add(rng, dtype, format):
+    if format == "coo":
+        pytest.xfail(reason="https://github.com/llvm/llvm-project/issues/116012")
     SHAPE = (100, 50)
     DENSITY = 0.5
     sampler = generate_sampler(dtype, rng)
+    sps_arr1 = sps.random_array(
+        SHAPE, density=DENSITY, format=format, dtype=dtype, random_state=rng, data_sampler=sampler
+    )
+    sps_arr2 = sps.random_array(
+        SHAPE, density=DENSITY, format=format, dtype=dtype, random_state=rng, data_sampler=sampler
+    )
```

Comment:
Can we still test adding different formats? `CSR+CSC`, `CSR+Dense`? We had it previously in the test.

Reply:
Done in https://github.com/pydata/sparse/pull/808/commits/0b910b93f6e12ba74bc8d7bdff7b7b5f6d220794. As a side-note, almost every case with COO segfaulted with `add`; so I really do think it is `add` failing on COO and not `asformat`.